src/python/pants/backend/docker/goals/run_image.py

```diff
@@ -3,27 +3,51 @@
 
 from __future__ import annotations
 
+from dataclasses import dataclass, replace
 from typing import cast
 
 from pants.backend.docker.goals.package_image import BuiltDockerImage, DockerFieldSet
 from pants.backend.docker.subsystems.docker_options import DockerOptions
+from pants.backend.docker.target_types import DockerImageRegistriesField, DockerImageSourceField
 from pants.backend.docker.util_rules.docker_binary import DockerBinary
 from pants.core.goals.package import BuiltPackage, PackageFieldSet
-from pants.core.goals.run import RunDebugAdapterRequest, RunRequest
+from pants.core.goals.run import RunDebugAdapterRequest, RunFieldSet, RunRequest
 from pants.engine.env_vars import EnvironmentVars, EnvironmentVarsRequest
 from pants.engine.rules import Get, MultiGet, collect_rules, rule
+from pants.engine.target import WrappedTarget, WrappedTargetRequest
+from pants.engine.unions import UnionRule
+
+
+@dataclass(frozen=True)
+class DockerRunFieldSet(RunFieldSet):
+    required_fields = (DockerImageSourceField,)
 
 
 @rule
 async def docker_image_run_request(
-    field_set: DockerFieldSet,
+    field_set: DockerRunFieldSet,
     docker: DockerBinary,
     options: DockerOptions,
     options_env_aware: DockerOptions.EnvironmentAware,
 ) -> RunRequest:
+    wrapped_target = await Get(
+        WrappedTarget,
+        WrappedTargetRequest(field_set.address, description_of_origin="<infallible>"),
+    )
+    build_request = DockerFieldSet.create(wrapped_target.target)
+    registries = options.registries()
+    for registry in registries.get(*(build_request.registries.value or [])):
+        if registry.run_as_alias:
+            # We only need to tag a single image name for run requests if there is a registry with
+            # `run_as_alias` as true.
+            build_request = replace(
```

Comment:
This is arguably a little bit hackish, but we simply override the list of registries selected for the docker build field set in case of a run request to only include the registry alias as a hardcoded value, overriding any defaults or field values. Sneaky.